src/test/java/uk/gov/pay/api/validation/PaymentRequestValidatorTest.java

```diff
@@ -13,34 +13,80 @@
-        CreatePaymentRequest paymentRequest = new CreatePaymentRequest(1, "https://return.to/return_url", "reference", "description");
+    public void validateMinimumAmount_shouldFailValue() throws Exception {
+        CreatePaymentRequest createPaymentRequest = createPaymentRequest(PaymentRequestValidator.AMOUNT_MIN_VALUE - 1, VALID_RETURN_URL, VALID_REFERENCE, VALID_DESCRIPTION);
+        expectedException.expect(ValidationException.class);
+        paymentRequestValidator.validate(createPaymentRequest);
     }
 
     @Test
-    public void validateMinimumAmount_shouldFailValue() throws Exception{
-        PaymentRequestValidator paymentRequestValidator = new PaymentRequestValidator(URLValidator.SECURITY_ENABLED);
-        CreatePaymentRequest paymentRequest = new CreatePaymentRequest(PaymentRequestValidator.AMOUNT_MAX_VALUE + 1, "https://return.to/return_url", "reference", "description");
+    public void validateMaximumAmount_shouldSuccessValue() {
+        CreatePaymentRequest createPaymentRequest = createPaymentRequest(PaymentRequestValidator.AMOUNT_MAX_VALUE, VALID_RETURN_URL, VALID_REFERENCE, VALID_DESCRIPTION);
+        paymentRequestValidator.validate(createPaymentRequest);
+    }
+
+    @Test
+    public void validateMaximumAmount_shouldFailValue() throws Exception {
+        CreatePaymentRequest paymentRequest = createPaymentRequest(PaymentRequestValidator.AMOUNT_MAX_VALUE + 1, VALID_RETURN_URL, VALID_REFERENCE, VALID_DESCRIPTION);
         expectedException.expect(ValidationException.class);
         paymentRequestValidator.validate(paymentRequest);
     }
 
     @Test
-    public void validateMaximumAmount_shouldSuccessValue(){
-        PaymentRequestValidator paymentRequestValidator = new PaymentRequestValidator(URLValidator.SECURITY_ENABLED);
-        CreatePaymentRequest paymentRequest = new CreatePaymentRequest(100, "https://return.to/return_url", "reference", "description");
+    public void validateReturnUrlMaxLength_shouldFailValue() throws Exception {
+        String invalidMaxLengthReturnUrl = "https://" + randomAlphanumeric(PaymentRequestValidator.URL_MAX_LENGTH) + ".com/";
+        CreatePaymentRequest paymentRequest = createPaymentRequest(VALID_AMOUNT, invalidMaxLengthReturnUrl, VALID_REFERENCE, VALID_DESCRIPTION);
+        expectedException.expect(ValidationException.class);
+        paymentRequestValidator.validate(paymentRequest);
     }
 
     @Test
-    public void validateMaximumAmount_shouldFailValue() throws Exception{
-        PaymentRequestValidator paymentRequestValidator = new PaymentRequestValidator(URLValidator.SECURITY_ENABLED);
-        CreatePaymentRequest paymentRequest = new CreatePaymentRequest(PaymentRequestValidator.AMOUNT_MAX_VALUE + 1, "https://return.to/return_url", "reference", "description");
+    public void validateReturnUrlSecurityEnabled_shouldFailValue() throws Exception {
```

Comment:
Can we have a more precise name like `validateReturnUrlNotHttps_shouldFailValue()` or something?